test-harness/specs/validate-body-params/json-inference-invalid-request.pc.txt

```diff
@@ -0,0 +1,85 @@
+====test====
+When two examples of the same request were supplied to Prism
+and there is a difference between request bodies
+Prism can reject the request which is not compatible with any of example bodies
+====spec====
+{
+  "info": {
+    "name": "Test"
+  },
+  "item": [
+    {
+      "id": "3b4f5a70-2a8f-46ba-a04b-a6f80621ad3f",
+      "request": {
+        "url": {
+          "path": ["path"]
+        },
+        "header": [
+          {
+            "key": "Content-Type",
+            "value": "application/json"
+          }
+        ],
+        "body": {
+          "mode": "raw",
+          "raw": "{\"to\": \"344455434\",\"type\": \"image\",\"recipient_type\": \"individual\",\"image\": {\"caption\":\"Quo Vadis\",\"id\":\"2bd2826a-c9ed-431d-beb6-f0eed5c24382\"}}"
+        },
+        "method": "POST"
+      },
+      "response": [
+        {
+          "code": 200,
+          "description": "OK",
+          "header": [
+            {
+              "key": "Content-Type",
+              "value": "text/plain"
+            }
+          ],
+          "body": "ok"
+        }
+      ]
+    },
+    {
+      "id": "3b4f5a70-2a8f-46ba-a04b-a6f80621ad3e",
+      "request": {
+        "url": {
+          "path": ["path"]
+        },
+        "header": [
+          {
+            "key": "Content-Type",
+            "value": "application/json"
+          }
+        ],
+        "body": {
+          "mode": "raw",
+          "raw": "{\"to\":\"342334535335\",\"type\":\"text\",\"recipient_type\":\"individual\",\"text\": {\"body\":\"Welcome\"}}"
+        },
+        "method": "POST"
+      },
+      "response": [
+        {
+          "code": 200,
+          "description": "OK",
+          "header": [
+            {
+              "key": "Content-Type",
+              "value": "text/plain"
+            }
+          ],
+          "body": "ok"
+        }
+      ]
+    }
+  ]
+}
+====server====
+mock -p 4010 ${document}
+====command====
+curl -i -X POST http://localhost:4010/path -H "Content-Type: application/json" --data '{"to":"123","type":"image","recipient_type":"individual"}'
+====expect====
+HTTP/1.1 422 Unprocessable Entity
+content-type: application/problem+json
+
+{"type":"https://stoplight.io/prism/errors#UNPROCESSABLE_ENTITY","title":"Invalid request body payload","status":422,"detail":"Your request is not valid and no HTTP validation response was found in the spec, so Prism is generating this error for you.","validation":[{"location":["body"],"severity":"Error","code":"required","message":"should have required property 'image'"},{"location":["body"],"severity":"Error","code":"required","message":"should have required property 'text'"},{"location":["body"],"severity":"Error","code":"anyOf","message":"should match some schema in anyOf"}]}
```

Comment:
The error message is a little bit confusing — I guess it's because of the way we construct the schema. In any case, we can iterate on this successively.

Reply:
Yes, that's because of how underlying schema is constructed.
Anyhow, it is also correct: the body should have either image or text defined in order to be well-formed.